tests/trainer/test_dataloaders.py

```diff
@@ -1227,21 +1227,22 @@ def validation_step(self, batch, batch_idx):
 
     trainer.test()
 
-    assert len(trainer.dev_debugger.val_dataloader_calls) == 3
+    assert len(trainer.dev_debugger.val_dataloader_calls) == 4
     assert len(trainer.dev_debugger.train_dataloader_calls) == 3
     assert len(trainer.dev_debugger.test_dataloader_calls) == 1
 
     # verify the sequence
     calls = trainer.dev_debugger.dataloader_sequence_calls
 
     expected_sequence = [
-        "train_dataloader",
-        "val_dataloader",
-        "train_dataloader",
-        "val_dataloader",
-        "train_dataloader",
-        "val_dataloader",
-        "test_dataloader",
+        'train_dataloader',
+        'val_dataloader',
+        'val_dataloader',
```

Comment:
this is an extra reload coming from the training loop calling run_evaluation. we could special case this by avoiding the reload if epoch = 0 but that feels hacky too: https://github.com/PyTorchLightning/pytorch-lightning/blob/68eac4d9486b95f8d1b09ae5468885c48d3d5cd0/pytorch_lightning/trainer/evaluation_loop.py#L63-L64

Reply:
We are doing this for train already: https://github.com/PyTorchLightning/pytorch-lightning/blob/44d775fccfb825561937f6fa03fe258af25c2b83/pytorch_lightning/trainer/training_loop.py#L169

Reply:
@tchaton does checking `trainer.current_epoch` work if someone is running multiple validation epochs per training epoch? is there a risk that we don't reload the dataloaders on each new validation pass?

Reply:
Yes, I think there is a risk on that. We could add a separate counter for validation epochs though

Reply:
can you add a comment to the code with this explanation?